### PR TITLE
fix!: Use SHA-256 for the config file name

### PR DESCRIPTION
**Pull request by silverwind**
Fixed a maximum line length eslint warning on this line as well. This change doubles the length of the file name from a previous 32 hex characters to 64.

SHA-256 was added in OpenSSL 0.9.8, released 2005-06-06, so every Node.js version should have it.

Fixes https://github.com/gulpjs/v8flags/issues/56

### CODE REVIEW

**Comment by phated** [comment]
One thing that we need to be conscious of for this is that the increase in filename length might cause Windows systems to fail with a max filepath limit.

We might want to release this as a breaking change due to that ☝️

**Comment by silverwind** [comment]
Indeed, thought I do consider it unlikely we'll reach `MAX_PATH` of 260 chars. In the general case the path will be something like

```
C:\Users\<username>\AppData\Local\js-v8flags\.v8flags-1-8.1.307.31-node.33.<hash>.json
```

which is around 135 charts + length of the username, so should be well within that limit.

**Comment by tgflet** [comment]
Bump, I could really use this fix in a project. It is breaking the project when enforcing FIPS when enforcing FIPS because MD5 is not FIPS compliant @silverwind @phated

**Comment by Shawn1874** [comment]
What is the status of this pull request?  I need this change for a project due to FIPS compliance issues.  It's been about a year since it was reported.  Can someone provide an estimate for when this will be merged and available?  Alternatively does anyone know of a work around?  Is there a way I can configure NPM to override the use of the MD5 hash prior to this pull request being accepted?

**Comment by phated** [comment]
@Shawn1874 As with all open source, it'll get done when it gets done. It wasn't until recently that gulp started to have any significant sponsors since all funding disappeared since covid.

**Comment by Shawn1874** [comment]
No, that is not how all open source projects work @phated.  Every project tends to involve different stakeholders and development pace.  I'd appreciate a less condescending response.  The pull request was submitted over a year with the changes, so I was just curious if someone could shed some light on why it wasn't accepted, and if there is a work around.

File: index.js [hunk 1]
```diff
@@ -15,14 +15,8 @@ var exclusions = ['--help', '--completion_bash'];
 // This number must be incremented whenever the generated cache file changes.
 var CACHE_VERSION = 2;
```
Comment (phated) [inline index.js:16]: Let's bump this too.
```suggestion
var CACHE_VERSION = 3;
```

**Comment by phated** [comment]
Thanks for this @silverwind! Sorry for the delay, but wanted to include this in the next major in case it breaks someone.

**Pull request merged**